### ISSUE TITLE
Nullable structs processing

### ISSUE THREAD

**Issue by incoboggart** [aaubry/YamlDotNet#360]
Hi!

I have an issue with YamlDotNet and dont understand how to resolve it.
The code is:

```
public class SomeClass
{
    public SomeStruct? Struct { get; set; }
}

public struct SomeStruct
{
    public int Value { get; set; }
}

public static void Test()
    {
        var s = new YamlDotNet.Serialization.SerializerBuilder().Build();

        var yaml = s.Serialize(new SomeClass
        {
            Struct = new SomeStruct { Value = 42 }
        });

        var d = new YamlDotNet.Serialization.DeserializerBuilder().Build();
        var i = d.Deserialize<SomeClass>(yaml);
    }
```

This code is throwing exception:
SerializationException: Property 'Value' not found on type 'System.Nullable`1[[SomeStruct]]

Is there any workaround for parsing such structure?

Thanks!

**Comment by cgranade** [comment]
I'm running into the same issue on 8.1.2:

![image](https://user-images.githubusercontent.com/31516/83801281-b8c04880-a65d-11ea-9549-668553fa79a7.png)

**Comment by benrebak** [comment]
I'm also running into this issue. Any advice?

**Comment by aaubry** [comment]
A fix for this issue has been released in [version 11.2.1](https://github.com/aaubry/YamlDotNet/releases/tag/v11.2.1).

**Comment by EdwardCooke** [comment]
Since there was a fix released for this a long time ago I am closing it.
